sql/parser/builtins.go

```diff
@@ -347,6 +348,80 @@ var builtins = map[string][]builtin{
 	"max": aggregateImpls(boolType, intType, floatType, stringType),
 	"min": aggregateImpls(boolType, intType, floatType, stringType),
 	"sum": aggregateImpls(intType, floatType),
+
+	// Timestamp/Date functions.
+
+	"age": {
+		builtin{
+			types:      typeList{timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: time.Now().Sub(args[0].(DTimestamp).Time)}, nil
+			},
+		},
+		builtin{
+			types:      typeList{timestampType, timestampType},
+			returnType: DummyInterval,
+			fn: func(args DTuple) (Datum, error) {
+				return DInterval{Duration: args[0].(DTimestamp).Sub(args[1].(DTimestamp).Time)}, nil
+			},
+		},
+	},
+
+	"current_date": {
+		builtin{
+			types:      typeList{},
+			returnType: DummyDate,
```

Comment:
```
⏚ [tamird:~/src/go/src/github.com/cockroachdb/cockroach] master ± git grep -A10 -B15 -F .impure
sql/parser/normalize.go-
sql/parser/normalize.go-var _ Visitor = &isConstVisitor{}
sql/parser/normalize.go-
sql/parser/normalize.go-func (v *isConstVisitor) Visit(expr Expr, pre bool) (Visitor, Expr) {
sql/parser/normalize.go-        if pre && v.isConst {
sql/parser/normalize.go-                if isVar(expr) {
sql/parser/normalize.go-                        v.isConst = false
sql/parser/normalize.go-                        return nil, expr
sql/parser/normalize.go-                }
sql/parser/normalize.go-
sql/parser/normalize.go-                switch t := expr.(type) {
sql/parser/normalize.go-                case *Subquery:
sql/parser/normalize.go-                        v.isConst = false
sql/parser/normalize.go-                        return nil, expr
sql/parser/normalize.go-                case *FuncExpr:
sql/parser/normalize.go:                        // typeCheckFuncExpr populates t.fn.impure.
sql/parser/normalize.go:                        if _, err := typeCheckFuncExpr(t); err != nil || t.fn.impure {
sql/parser/normalize.go-                                v.isConst = false
sql/parser/normalize.go-                                return nil, expr
sql/parser/normalize.go-                        }
sql/parser/normalize.go-                }
sql/parser/normalize.go-        }
sql/parser/normalize.go-        return v, expr
sql/parser/normalize.go-}
sql/parser/normalize.go-
sql/parser/normalize.go-// isConst returns true if the expression contains only constant values
sql/parser/normalize.go-// (i.e. it does not contain a DReference).
```

The interesting bit is:
```go
case *FuncExpr:
        // typeCheckFuncExpr populates t.fn.impure.
        if _, err := typeCheckFuncExpr(t); err != nil || t.fn.impure {
                v.isConst = false
                return nil, expr
        }
}
```
`fn.impure` is true when a function is not pure, that is, it produces different results with the same inputs. This is used by `isConstVisitor` to determine that a function call with constant arguments is not a constant.